scanpy/preprocessing/_highly_variable_genes.py

```diff
@@ -338,12 +369,123 @@ def highly_variable_genes(
             ('dispersions_norm', 'float32'),
         ]
         if batch_key is not None:
-            arrays.extend([
-                df['highly_variable_nbatches'].values,
-                df['highly_variable_intersection'].values,
-            ])
-            dtypes.append([
-                ('highly_variable_nbatches', int),
-                ('highly_variable_intersection', np.bool_),
-            ])
+            arrays.extend(
+                [
+                    df['highly_variable_nbatches'].values,
+                    df['highly_variable_intersection'].values,
+                ]
+            )
+            dtypes.append(
+                [
+                    ('highly_variable_nbatches', int),
+                    ('highly_variable_intersection', np.bool_),
+                ]
+            )
         return np.rec.fromarrays(arrays, dtype=dtypes)
+
+
+def highly_variable_genes_seurat_v3(
+    adata: AnnData,
+    n_top_genes: int = 2000,
+    batch_key: Optional[str] = None,
+    lowess_frac: Optional[float] = 0.15,
+):
+    """\
+    Annotate highly variable genes [Stuart19]_.
+
+    Expects raw count data.
+
+    The major difference in this implementation is the use of lowess insted of loess.
+
+    For further details of the sparse arithmetic see https://www.overleaf.com/read/ckptrbgzzzpg
+    Parameters
+    ----------
+    adata
+        The annotated data matrix of shape `n_obs` × `n_vars`. Rows correspond
+        to cells and columns to genes.
+    n_top_genes
+        Number of highly-variable genes to keep.
+    batch_key
+        If specified, highly-variable genes are selected within each batch separately and merged.
+        This simple process avoids the selection of batch-specific genes and acts as a
+        lightweight batch correction method.
+    lowess_frac
+        The fraction of the data (cells) used when estimating the variance in the lowess model fit.
+    """
+    import statsmodels
+
+    lowess = statsmodels.nonparametric.lowess
```

Comment:
I don't think this import will work. [`statsmodels` import api is a bit strange](https://www.statsmodels.org/stable/api-structure.html#import-paths-and-structure), but I think this should work:

```suggestion
    from statsmodels.nonparametric.smoothers_lowess import lowess
```